IllegalArgumentException when invoking remote prosyst generator from toolset

Steps to reproduce:
1) In Eclipse Toolset, choose XDK in the model browser
2) Right click "Generate"
3) Generators Pop up appears
4) Choose Generate for ProSyst Generator

Expected :
Eclipse Project is generated containing ProSyst artefacts

Actual:
[ERROR] Error querying remote repository with generateCode request. [ IllegalArgumentException:Illegal character in path at index 114: http://vorto.eclipse.org/rest/generation-router/com.mycompany.informationmodels/DemoInformationModel/1.0.0/ProSyst FunctionalItem Generator ]

--- a/bundles/org.eclipse.vorto.repository/src/org/eclipse/vorto/repository/RestModelRepository.java
+++ b/bundles/org.eclipse.vorto.repository/src/org/eclipse/vorto/repository/RestModelRepository.java
@@ -194,7 +194,6 @@ public class RestModelRepository extends Observable implements IModelRepository 
 		try {
 			String url = "generation-router/" + model.getNamespace() + "/" + model.getName() + "/" + model.getVersion()
 					+ "/" + URLEncoder.encode(serviceKey, "utf-8");
-			System.out.println("-erle- : " +  url);
 			Attachment result = httpClient.executeGetAttachment(url);
 			return result;
 		} catch (RepositoryException e) {

--- a/bundles/org.eclipse.vorto.repository/src/org/eclipse/vorto/repository/RestModelRepository.java
+++ b/bundles/org.eclipse.vorto.repository/src/org/eclipse/vorto/repository/RestModelRepository.java
@@ -193,7 +193,8 @@ public class RestModelRepository extends Observable implements IModelRepository 
 	public Attachment generateCode(ModelId model, String serviceKey) {
 		try {
 			String url = "generation-router/" + model.getNamespace() + "/" + model.getName() + "/" + model.getVersion()
-					+ "/" + serviceKey;
+					+ "/" + URLEncoder.encode(serviceKey, "utf-8");
+			System.out.println("-erle- : " +  url);
 			Attachment result = httpClient.executeGetAttachment(url);
 			return result;
 		} catch (RepositoryException e) {

--- a/server/generators/org.eclipse.vorto.codegen.service.prosystfi/pom.xml
+++ b/server/generators/org.eclipse.vorto.codegen.service.prosystfi/pom.xml
@@ -19,6 +19,10 @@
 		</dependency>
 		<dependency>
 			<groupId>org.eclipse.vorto</groupId>
+			<artifactId>org.eclipse.vorto.codegen.templates.java</artifactId>
+		</dependency>				
+		<dependency>
+			<groupId>org.eclipse.vorto</groupId>
 			<artifactId>org.eclipse.vorto.codegen.examples.prosystfi</artifactId>
 			<version>${project.version}</version>
 		</dependency>

--- a/server/pom.xml
+++ b/server/pom.xml
@@ -69,6 +69,11 @@
 			</dependency>
 			<dependency>
 				<groupId>org.eclipse.vorto</groupId>
+				<artifactId>org.eclipse.vorto.codegen.templates.java</artifactId>
+				<version>${project.version}</version>
+			</dependency>
+			<dependency>
+				<groupId>org.eclipse.vorto</groupId>
 				<artifactId>org.eclipse.vorto.editor</artifactId>
 				<version>${project.version}</version>
 			</dependency>

--- a/server/repo/repo-ui/src/main/java/org/eclipse/vorto/repository/web/ModelGenerationController.java
+++ b/server/repo/repo-ui/src/main/java/org/eclipse/vorto/repository/web/ModelGenerationController.java
@@ -83,8 +83,6 @@ public class ModelGenerationController extends RepositoryController {
 		request.getParameterMap().entrySet().stream().forEach(x -> requestParams.put(x.getKey(), x.getValue()[0]));
 		
 		try {
-			System.out.println("-erle- : " + URLDecoder.decode(serviceKey, "utf-8"));
-			
 			GeneratedOutput generatedOutput = generatorService.generate(new ModelId(name,namespace,version), URLDecoder.decode(serviceKey, "utf-8"), requestParams);
 			response.setHeader(CONTENT_DISPOSITION, ATTACHMENT_FILENAME + generatedOutput.getFileName());
 			response.setContentLengthLong(generatedOutput.getSize());

--- a/server/repo/repo-ui/src/main/java/org/eclipse/vorto/repository/web/ModelGenerationController.java
+++ b/server/repo/repo-ui/src/main/java/org/eclipse/vorto/repository/web/ModelGenerationController.java
@@ -16,6 +16,7 @@ package org.eclipse.vorto.repository.web;
 
 import java.io.ByteArrayInputStream;
 import java.io.IOException;
+import java.net.URLDecoder;
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.HashMap;
@@ -80,12 +81,15 @@ public class ModelGenerationController extends RepositoryController {
 
 		Map<String, String> requestParams = new HashMap<>();
 		request.getParameterMap().entrySet().stream().forEach(x -> requestParams.put(x.getKey(), x.getValue()[0]));
-		 
-		GeneratedOutput generatedOutput = generatorService.generate(new ModelId(name,namespace,version), serviceKey, requestParams);
-		response.setHeader(CONTENT_DISPOSITION, ATTACHMENT_FILENAME + generatedOutput.getFileName());
-		response.setContentLengthLong(generatedOutput.getSize());
-		response.setContentType(APPLICATION_OCTET_STREAM);
+		
 		try {
+			System.out.println("-erle- : " + URLDecoder.decode(serviceKey, "utf-8"));
+			
+			GeneratedOutput generatedOutput = generatorService.generate(new ModelId(name,namespace,version), URLDecoder.decode(serviceKey, "utf-8"), requestParams);
+			response.setHeader(CONTENT_DISPOSITION, ATTACHMENT_FILENAME + generatedOutput.getFileName());
+			response.setContentLengthLong(generatedOutput.getSize());
+			response.setContentType(APPLICATION_OCTET_STREAM);
+		
 			IOUtils.copy(new ByteArrayInputStream(generatedOutput.getContent()), response.getOutputStream());
 			response.flushBuffer();
 		} catch (IOException e) {

